commit-event-service/src/test/scala/io/renku/commiteventservice/events/categories/common/CommitInfoFinderSpec.scala

```diff
@@ -143,110 +129,90 @@ class CommitInfoFinderSpec
+        val someCommitInfoExpectation = CommitInfo(
+          id = commitId,
+          message = commitMessage,
+          committedDate = committedDate,
+          author = author,
+          committer = committer,
+          parents = parents
+        ).some
+
+        setGitLabClientExpectation(maybeAccessToken, returning = someCommitInfoExpectation)
+
+        finder
+          .getMaybeCommitInfo(projectId, commitId)(maybeAccessToken)
+          .unsafeRunSync() shouldBe someCommitInfoExpectation
       }
 
     "get commit info from the configured url " +
       "and return some CommitInfo if OK returned with valid body - case with no access token" in new TestCase {
 
-        stubFor {
-          get(s"/api/v4/projects/$projectId/repository/commits/$commitId")
-            .willReturn(okJson(responseJson.toString()))
-        }
-
-        finder.getMaybeCommitInfo(projectId, commitId)(maybeAccessToken = None).unsafeRunSync() shouldBe common
-          .CommitInfo(
-            id = commitId,
-            message = commitMessage,
-            committedDate = committedDate,
-            author = author,
-            committer = committer,
-            parents = parents
-          )
-          .some
-      }
+        val someCommitInfoExpectation = CommitInfo(
+          id = commitId,
+          message = commitMessage,
+          committedDate = committedDate,
+          author = author,
+          committer = committer,
+          parents = parents
+        ).some
 
-    "return None if remote client responds with Not found" in new TestCase {
+        setGitLabClientExpectation(maybeAccessToken, returning = someCommitInfoExpectation)
 
-      stubFor {
-        get(s"/api/v4/projects/$projectId/repository/commits/$commitId")
-          .willReturn(notFound())
+        finder
+          .getMaybeCommitInfo(projectId, commitId)(maybeAccessToken)
+          .unsafeRunSync() shouldBe someCommitInfoExpectation
       }
+  }
 
-      finder.getMaybeCommitInfo(projectId, commitId).unsafeRunSync() shouldBe None
+  "return None if remote client responds with Not found" in new TestCase {
+    mapToMaybeCommit((Status.NotFound, Request[IO](), Response[IO]())).unsafeRunSync() shouldBe None
+  }
 
-    }
+  // TODO: figure out how to check that it's called twice
+  "fallback to fetch commit info without an access token for UNAUTHORIZED" in new TestCase {
+    val invalidAccessToken = accessTokens.generateSome
 
-    "fallback to fetch commit info without an access token for UNAUTHORIZED" in new TestCase {
+    val mapResponse = CaptureOne[ResponseMappingF[IO, Option[CommitInfo]]]()
 
-      stubFor {
-        get(s"/api/v4/projects/$projectId/repository/commits/$commitId")
-          .withAccessToken(maybeAccessToken)
-          .willReturn(unauthorized())
-      }
-      stubFor {
-        get(s"/api/v4/projects/$projectId/repository/commits/$commitId")
-          .willReturn(okJson(responseJson.toString()))
-      }
+    (gitLabClient
```

Comment:
I saw that you managed to write a similar test in a nicer way in the LatestCommitFetcherSpec in the _fallback to fetching the latest commit without an access token for UNAUTHORIZED_ test